propertyestimator/protocols/simulation.py

```diff
@@ -394,33 +401,36 @@ def _write_statistics_array(self, raw_statistics, temperature, pressure,
         StatisticsArray
             The statistics array with statistics appended.
         """
-        beta = 1.0 / (unit.BOLTZMANN_CONSTANT_kB * temperature)
+        temperature = openmm_quantity_to_pint(temperature)
+        pressure = openmm_quantity_to_pint(pressure)
+
+        beta = 1.0 / (unit.boltzmann_constant * temperature)
 
-        potential_energies = np.array(raw_statistics[ObservableType.PotentialEnergy]) * unit.kilojoule_per_mole
-        kinetic_energies = np.array(raw_statistics[ObservableType.KineticEnergy]) * unit.kilojoule_per_mole
+        potential_energies = np.array(raw_statistics[ObservableType.PotentialEnergy]) * unit.kilojoules / unit.mole
+        kinetic_energies = np.array(raw_statistics[ObservableType.KineticEnergy]) * unit.kilojoules / unit.mole
         volumes = np.array(raw_statistics[ObservableType.Volume]) * unit.angstrom ** 3
 
         # Calculate the instantaneous temperature, taking account the
         # systems degrees of freedom.
-        temperatures = 2.0 * kinetic_energies / (degrees_of_freedom * unit.MOLAR_GAS_CONSTANT_R)
+        temperatures = 2.0 * kinetic_energies / (degrees_of_freedom * unit.molar_gas_constant)
 
         # Calculate the systems enthalpy and reduced potential.
         total_energies = potential_energies + kinetic_energies
         enthalpies = None
 
-        reduced_potentials = potential_energies / unit.AVOGADRO_CONSTANT_NA
+        reduced_potentials = potential_energies / unit.avogadro_number
```

Comment:
(Not a suggestion for this PR, but maybe a discussion we should have about standardizing design)
I'm worried that we're going to reach a reckoning some day with the fact that we use `kJ/mole` to mean two completely separate things:
1) A very small amount of energy, like "a microfemtojoule (or whatever the proper prefix is)", that's useful for describing amounts of energy like those that a molecule encounters. 
2) An actual unit of energy divided by a unit of quantity, like "this dynamite has 100 kJ/mol of energy. If you add more moles, you get more energy."